lib/luanetfilter.c

```diff
@@ -43,6 +73,67 @@ static void luanetfilter_release(void *private)
 	lunatik_putobject(luaxtable->runtime);
 }
 
+/* Register the target with the netfilter module. The caller must push the table on the stack.
+ * Pops the table from the stack both on success and failure.
+ * Pushes the error string on the stack on failure and returns -1.
+ * */
+static int handle_xtable_reg(lua_State *L, luanetfilter_xtable_t *luaxtable, LUANETFILTER_MODE mode)
+{
+	int base = lua_gettop(L);
+	int type;
+
+	const char *name = NULL;
+	u_int8_t revision = 0;
+	unsigned short family = 0;
+	unsigned short proto = 0;
+
+	type = lua_type(L, base);
+	if (type == LUA_TNIL){
+		pr_err("[luanetfilter] no ops found, assuming nil\n");
+		return 0;
+	} else if (type != LUA_TTABLE){
+		lua_pushstring(L, "invalid ops type, expected table or nil");
+		goto err;
+	}
+
+	luanetfilter_xtable_op(L, base, "name", lua_isstring);
+	name = lua_tostring(L, -1);
+
+	luanetfilter_xtable_op(L, base, "revision", lua_isnumber);
+	revision = (u_int8_t)lua_tointeger(L, -1);
+
+	luanetfilter_xtable_op(L, base, "family", lua_isnumber);
+	family = (unsigned short)lua_tointeger(L, -1);
+
+	luanetfilter_xtable_op(L, base, "proto", lua_isnumber);
+	proto = (unsigned short)lua_tointeger(L, -1);
+
+	luanetfilter_xtable_op_func(L, base, "checkentry");
+	luanetfilter_xtable_op_func(L, base, "destroy");
+
+	if (mode == LUANETFILTER_MATCH)
+		luanetfilter_xtable_op_func(L, base, "match");
+	else
+		luanetfilter_xtable_op_func(L, base, "target");
+
+	/* Print for debugging */
+	pr_err("[luanetfilter] target name : %s\n", name);
+	pr_err("[luanetfilter] target revision : %u\n", revision);
+	pr_err("[luanetfilter] target family : %d\n", family);
+	pr_err("[luanetfilter] target proto : %d\n", proto);
```

Comment:
let's no add this to PRs

Reply:
yeah, I will remove these comments before getting the PR merged.